src/components/Authenticator.js

```diff
@@ -1,30 +1,42 @@
-import Cookies from 'js-cookie';
+import {findDocumentQuery} from './database/Queries';
+import type {OAuthUserType, UserType} from './admin/FlowTypes';
 
-const mockAuth = {
-  isAdmin: function() {
-    const adminCookie = Cookies.get('admin');
-    return (adminCookie !== undefined);
-  },
-
-  isUser: function() {
-    const userCookie = Cookies.get('user');
-    return (userCookie !== undefined);
-  },
+const readUserDataFromDB = (user) => {
+  if (user === null) {
+    return Promise.resolve([]);
+  }
+  return findDocumentQuery('active-members', 'email', user.email);
+};
 
-  isInviter: function() {
-    const inviterCookie = Cookies.get('inviter');
-    return (inviterCookie !== undefined);
+const firebaseAuthenticator = {
+  isAdmin: function(user: OAuthUserType): Promise<boolean> {
+    let userDataPromise = readUserDataFromDB(user);
+    if (userDataPromise) {
+      return userDataPromise.then(
+          (userData: Array<UserType>) =>
+              Promise.resolve(userData.length > 0 && userData[0].isAdmin)
+          );
+    } else {
+      return Promise.resolve(false);
+    }
   },
```

Comment:
this seems overly complicated. Also, I don't understand `if (cuserDataPromise)`: testing it, it seem to return always `true` for me.

Reply:
Yes, you are right. I simplified it. Previously I was thinking that DB returns null if the user doesn't exist. But it just returns empty Array, so all checks were unnecessary! Thank you for pointing it out :-)